rust/src/push/evaluator.rs

```diff
@@ -256,9 +256,11 @@ impl PushRuleEvaluator {
         };
 
         let result = match known_condition {
-            KnownCondition::EventMatch(event_match) => {
-                self.match_event_match(&event_match.key.clone(), &event_match.pattern.clone())?
-            }
+            KnownCondition::EventMatch(event_match) => self.match_event_match(
+                &self.flattened_keys,
+                &event_match.key.clone(),
```

Comment:
I'm surprised you need to clone here. `event_match.key` is a `Cow<'static, str>`; can you not pass `&event_match.key`? See https://stackoverflow.com/a/47159528/5252017

Reply:
It seems I don't! I thought it had complained about needing to own things when I did that, probably some refactoring somewhere else fixed it?